filter.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+const { config } = require('./config')
+
+const minimatch = require('minimatch')
+const path = require('path')
+
+/**
+ * Checks, whenever a file should be excluded by the,
+ * excludes provided by the user
+ * @param {String} filePath path to the file which will be checked
+ * @param {String[]} excludes array of exclude rules against whom
+ * the file will be checked
+ */
+function isExcluded (filePath, excludes) {
+  let ok = false
+  for (let excludeRule of excludes) {
+    excludeRule = path.normalize(excludeRule)
+    if (minimatch(filePath, excludeRule, { matchBase: true })) {
+      ok = true
+      break
+    }
+  }
+  return ok
+}
+
+/**
+ * Filters the listed files and returns only the files relevant to us
+ * @param {*} rawData the raw output from list files API call
+ * @param {String[] } excludes globs provided by the user to exclude files/folders
+ * @return {Promise[]<any>} the filtered GitHub response
+ */
+function filterData (rawData, excludes) {
+  return rawData.data
+    .filter(file =>
+      config.fileExtensions.reduce(
+        (acc, ext) => acc || file.filename.endsWith(ext),
+        false
+      )
+    )
+    .filter(fileJSON => fileJSON.status !== 'removed')
+    .map(fileInfo => {
+      return fileInfo.filename
+    })
+    .filter(filePath => {
+      if (excludes.length > 0) {
```

Comment:
I believe this check is redundant? Won't the `for` loop in `isExcluded` simply skip straight to returning the default, false, value for ok if we pass an empty array for excludes?

Reply:
Yep, I checked and tested and you are right.